src/Polysemy/Internal/Union.hs

```diff
@@ -286,21 +284,22 @@ weaken (Union pr a) = Union (There pr) a
 
 ------------------------------------------------------------------------------
 -- | Lift an effect @e@ into a 'Union' capable of holding it.
-inj :: forall e r m a. (Functor m , Member e r) => e m a -> Union r m a
+inj :: forall e r rInitial a. (Member e r) => e (Sem rInitial) a -> Union r (Sem rInitial) a
 inj e = injWeaving $
   Weaving e (Identity ())
-            (fmap Identity . runIdentity)
-            runIdentity
-            (Just . runIdentity)
+              (fmap Identity . runIdentity)
+              runIdentity
+              (Just . runIdentity)
```

Comment:
I don't like these extra indents: the arguments no longer line up. That said, the original was wrong too. See my comment above for how arguments to a `Weaving` should look like